cli/src/pcluster3_config_converter/pcluster3_config_converter.py

```diff
@@ -0,0 +1,1000 @@
+        """Convert HeadNode Imds section."""
+        imds = dict()
+        if self.cluster_config_get("scheduler") == "awsbatch":
+            imds["Secured"] = False
+        _add_if(headnode, "Imds", imds)
+
+    def convert_single_slurm_compute_resource(self, compute_resource_label, queue_section):
+        """Convert single compute resource for a slurm queue."""
+        compute_resource = dict()
+        efa = dict()
+        compute_resource_section = f"compute_resource {compute_resource_label.strip()}"
+        self.validate_single_field(compute_resource_section, "initial_count", "ignored")
+        compute_resource["Name"] = compute_resource_label.strip().replace("_", "-")
+        for item in [
+            (compute_resource_section, "instance_type", compute_resource, "InstanceType"),
+            (compute_resource_section, "min_count", compute_resource, "MinCount", "getint"),
+            (compute_resource_section, "max_count", compute_resource, "MaxCount", "getint"),
+            (compute_resource_section, "spot_price", compute_resource, "SpotPrice", "getfloat"),
+            (
+                queue_section,
+                "disable_hyperthreading",
+                compute_resource,
+                "DisableSimultaneousMultithreading",
+                "getboolean",
+            ),
+            (queue_section, "enable_efa", efa, "Enabled", "getboolean"),
+            (queue_section, "enable_efa_gdr", efa, "GdrSupport", "getboolean"),
+        ]:
+            self.convert_single_field(*item)
+        if self.config_parser.getboolean(queue_section, "disable_hyperthreading", fallback=None) is None:
+            self.convert_single_field(
+                self.cluster_section_name,
+                "disable_hyperthreading",
+                compute_resource,
+                "DisableSimultaneousMultithreading",
+                "getboolean",
+            )
+
+        _add_if(compute_resource, "Efa", efa)
+        return compute_resource
+
+
+def _parse_args(argv=None):
+    """Parse command line args."""
+    convert_parser = argparse.ArgumentParser(
+        description="Convert AWS ParallelCluster configuration file.",
+        epilog='For command specific flags, please run: "pcluster3-config-converter --help"',
+    )
+    convert_parser.add_argument(
+        "-c",
+        "--config-file",
+        help="Configuration file to be used as input.",
+    )
+    convert_parser.add_argument(
+        "-t",
+        "--cluster-template",
+        help=(
+            "Indicates the 'cluster' section of the configuration file to convert. "
+            "If not specified the script will look for the cluster_template parameter in the [global] section "
+            "or will search for '[cluster default]'."
+        ),
+        required=False,
+    )
+    convert_parser.add_argument(
+        "-o",
+        "--output-file",
+        help="Configuration file to be written as output. By default the output will be written to the stdout.",
```

Comment:
`written to stdout` (no `the` required here)

Reply:
Changed